cdap-app-templates/cdap-etl/cdap-etl-proto/src/main/java/io/cdap/cdap/etl/proto/v2/spec/StageSpec.java

```diff
@@ -118,6 +133,7 @@ public boolean equals(Object o) {
     return Objects.equals(name, that.name) &&
       Objects.equals(plugin, that.plugin) &&
       Objects.equals(inputSchemas, that.inputSchemas) &&
+      Objects.equals(Sets.newHashSet(inputStages), Sets.newHashSet(that.inputStages)) &&
```

Comment:
You can use the Standard Java API `new HashSet<>(inputStages)`.

Reply:
fixed